src/main/java/usecases/friendinteractors/acceptfriend/acceptFriendInteractor.java

```diff
@@ -0,0 +1,32 @@
+package usecases.friendinteractors.acceptfriend;
+
+public class acceptFriendInteractor implements acceptFriendInputBoundary {
+    final acceptFriendOutputBoundary output;
+    final acceptFriendDSGateway dataBase;
+
+
+    public acceptFriendInteractor(acceptFriendDSGateway dataBase, acceptFriendOutputBoundary output) {
+        this.dataBase = dataBase;
+        this.output = output;
+    }
+
+    /**
+     * the whole accept friend should be re-designed because it is a click from friend not a search from requester
+     * @param model
+     */
+    @Override
+    public void acceptFriend(acceptFriendInputModel model) {
+        int friendid = model.getFriendid();
+        int requesterid = model.getRequesterid();
+        String friendName = model.getFriendName();
+        int requesterPeerPort = dataBase.getPeerPort(requesterid);
+        String requesterAddress = dataBase.getAddress(requesterid);
+        int friendPeerPort = dataBase.getPeerPort(friendid);
+        String friendAddress = dataBase.getAddress(friendid);
+
+        dataBase.acceptFriendbyID(friendid, requesterid);
+        dataBase.refuseFriendbyID(friendid, requesterid);
```

Comment:
Not sure about the logic here, if someone is accepting another's friend request, shouldn't there be a (friendAccepted) input to the usecase, and then depending on the True/False of the input, the corresponding method will be called (i.e. acceptFriendbyID or refuseFriendbyID)